[Shopify] Connector - Company Location creation from Customer

AndriusA-Companial opened the pull request:
This pull request does not have a related issue as it's part of the delivery for development agreed directly with @AndreiPanko

- Added “Add customer as Shopify Location” action
- Business logic to handle creation of company location from customer added
- Added new Boolean field 'Manual Payment Gateway' field to 'Order Transaction' and 'Payment Method Mapping' tables.

Automated tests implemented

jzaksauskas committed:
Added Sell-to Bill-to fields and those fields editable in Company Locations page

jzaksauskas committed:
Marked possible implementation place with TODO

jzaksauskas committed:
Implemented Shopify Sell-To Bill-To mapping based on Company Location setup values

jzaksauskas committed:
Fixed RunPageLink for Shopify Orders action in Companies page

jzaksauskas committed:
Merge branch 'main' into dev/jza/comp_loc_cust_mapping

AndriusAndrulevicius committed:
Company location mapping to customer

GediminasGaubys commented on Apps/W1/Shopify/app/src/Companies/Tables/ShpfyCompanyLocation.Table.al:
```diff
@@ -117,6 +119,38 @@ table 30151 "Shpfy Company Location"
+        field(18; "Sell-to Customer No."; Code[20])
+        {
+            Caption = 'Sell-to Customer No.';
+            DataClassification = CustomerContent;
```
I don't know why in other fields DataClassification is declared, but since it is declared at the top, no need to repeat it in the fields.

GediminasGaubys commented on Apps/W1/Shopify/app/src/Companies/Pages/ShpfyCompLocations.Page.al:
```diff
@@ -1,4 +1,5 @@
 namespace Microsoft.Integration.Shopify;
+using Microsoft.Sales.Customer;
```
add an empty line after namespace.

GediminasGaubys commented on Apps/W1/Shopify/app/src/Helpers/Codeunits/ShpfyJsonHelper.Codeunit.al:
```diff
@@ -286,6 +286,14 @@ codeunit 30157 "Shpfy Json Helper"
+
+    internal procedure IsTokenNull(JToken: JsonToken): Boolean
+    begin
+        if JToken.IsValue then
```
add IsValue**()**

GediminasGaubys commented on Apps/W1/Shopify/app/src/Helpers/Codeunits/ShpfyJsonHelper.Codeunit.al:
```diff
@@ -286,6 +286,14 @@ codeunit 30157 "Shpfy Json Helper"
+    internal procedure IsTokenNull(JToken: JsonToken): Boolean
+    begin
+        if JToken.IsValue then
+            exit(JToken.AsValue().IsNull)
```
Add ()

AndriusAndrulevicius committed:
Missing automated test codeunit added

GediminasGaubys commented on Apps/W1/Shopify/test/Companies/ShpfyCompanyLocationsTest.Codeunit.al:
```diff
@@ -0,0 +1,143 @@
+    TestPermissions = Disabled;
+    TestHttpRequestPolicy = BlockOutboundRequests;
+
+
```
remove empty line

AndriusAndrulevicius committed:
Comments resolved

jzaksauskas commented on Apps/W1/Shopify/test/Companies/ShpfyCompanyLocationsTest.Codeunit.al:
```diff
@@ -0,0 +1,142 @@
+    var
+        ShpfyCompany: Record "Shpfy Company";
+        ShpfyCompanyAPI: Codeunit "Shpfy Company API";
+        ShpfyCompanies: TestPage "Shpfy Companies";
```
do not use Test Pages

jzaksauskas commented on Apps/W1/Shopify/app/src/GraphQL/Codeunits/ShpfyGQLCreateCompLocation.Codeunit.al:
```diff
@@ -0,0 +1,27 @@
+    /// <returns>Return value of type Text.</returns>
+    internal procedure GetGraphQL(): Text
+    begin
+        exit('{"query": "mutation CompLocCreate { companyLocationCreate( companyId: \"gid://shopify/Company/{{CompanyId}}\" input: { externalId: \"{{ExternalId}}\" name: \"{{Name}}\" phone: \"{{Phone}}\" taxExempt: {{TaxExempt}} taxRegistrationId: \"{{TaxRegistrationId}}\" billingAddress: { address1: \"{{BillingAddress1}}\" address2: \"{{BillingAddress2}}\" city: \"{{BillingCity}}\" countryCode:  ...
```
'CompLocCreate ' not needed

AndriusAndrulevicius committed:
Comments fixed

AndriusAndrulevicius committed:
Add 'Manual Payment Gateway' field to Order Transaction and Payment Method Mapping tables; update GraphQL query to include it

AndriusAndrulevicius committed:
Refactor company location handling and add report for adding customers as locations

AndriusAndrulevicius committed:
Rename ShpfyCompany variable to ShopifyCompany for consistency; update related references in Company API and report

AndriusAndrulevicius committed:
Update Apps/W1/Shopify/test/Companies/ShpfyCompanyLocationsTest.Codeunit.al

Co-authored-by: Jesper Schulz-Wedde <JesperSchulz@users.noreply.github.com>

AndriusAndrulevicius committed:
Update Shopify Company API and related tests; improve company location handling and GraphQL query

onbuyuka committed:
Merge branch 'main' into dev/jza/comp_loc_cust_mapping

Groenbech96 committed:
Syncing with version 27.0.37019.0 (#28920)

Fixes
[AB#560539](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/560539)

---------

Co-authored-by: Magnus Hartvig Grønbech <magnushar@microsoft.com>

GediminasGaubys committed:
[Shopify] Connector - Staff to Salesperson mapping (#28743)

This pull request does not have a related issue as it's part of the
delivery for development agreed directly with
[@AndreiPanko](https://github.com/AndreiPanko)

* New functionality is only available for B2B stores. Hide (don’t use if
the store is not B2B, see example – Companies/ Catalogs)
* Create the new Shopify Staff Mapping page (similar to
Languages/Markets).
* Add action to read staff list. Allow user to map Staff entry to the
Salesperson/purchaser entry.
* When importing orders, for B2B stores also import Staff node (id
only). Only for B2B stores to avoid errors during calls placed by
non-plus stores.
* Add the Sales Person Code field to the Shopify Order header table/page
and initialize during mapping routine (similar to Map Customer / Payment
Method).
* When converting order from Shopify Order to Business Central sales
document populate the Sales Person in the created sales order, similar
to Payment Method.
* Out of scope – export sales person to Shopify as Staff

---------

Co-authored-by: Andrius Andrulevicius <andrius.andrulevicius@companial.com>
Co-authored-by: Alexander Holstrup <117829001+aholstrup1@users.noreply.github.com>

aholstrup1 committed:
Merge branch 'main' into dev/jza/comp_loc_cust_mapping
